editor/src/messages/tool/common_functionality/shape_editor.rs

```diff
@@ -212,8 +216,21 @@ impl ShapeEditor {
 		}
 	}
 
-	/// The opposing handle lengths.
-	pub fn opposing_handle_lengths(&self, document: &Document) -> HashMap<Vec<LayerId>, HashMap<u64, f64>> {
+	/// Mirror the opposing handles.
+	///
+	/// We will keep track of the opposing handle of a manipulator group when:
+	/// 1. Exactly one handle is selected.
+	/// 2. The anchor is not selected.
+	/// 3. If the handle selected has an opposing handle, we need manipulator group's angle mirroring to be true.
+	///
+	/// If (1) and (2) are true and there is no opposing handle, we create one
+	/// (irrespective of angle mirroring) and set angle mirroring to true.
+	///
+	/// We return the opposing handle lengths.
+	///
+	/// toggle_angle_mirroring decides whether the current angle mirroring will be toggled.
+	/// It is required as messages don't immediately change the manipulator groups.
+	pub fn mirror_opposing_handles(&self, document: &Document, toggle_angle_mirroring: bool, responses: &mut VecDeque<Message>) -> OpposingHandleLengths {
```

Comment:
As mentioned in the comment above, I pass a `toggle_angle_mirroring` to check whether the angle mirroring will be toggled. This is required as the angle mirroring won't be updated immediately (instead messages to toggle are added to `responses`).

Is this fine? Is there a better way to do this?

Reply:
This is fine.